website/static/js/signIn.js

```diff
@@ -5,59 +5,49 @@
 
 var ko = require('knockout');
 require('knockout-validation').init({insertMessages: false});  // override default DOM insertions
-var $ = require('jquery');
 
 var $osf = require('js/osfHelpers');
-
-var ViewModel = function() {
-
-    var self = this;
-
-    self.username = ko.observable('').extend({
-        required: true,
-        email: true
-    });
-    self.password = ko.observable('').extend({
-        required: true,
-        minLength: 6,
-        maxLength: 35
-    });
-
-    self.isValid = ko.pureComputed(function() {
-        return self.username.isValid() && self.password.isValid();
-    });
-
-    self.submit = function() {
-        // Show errors if invalid
-        if (!self.isValid()) {
-            if (!self.username.isValid()) {
-                $osf.growl(
-                    'Error',
-                    'Please enter a valid email address.',
-                    'danger'
-                );
-            }
-            if (!self.password.isValid()) {
-                $osf.growl(
-                    'Error',
-                    'Your password must be more than six characters.',
-                    'danger'
-                );
-            }
-            return false; // Stop form submission
+var oop = require('js/oop');
+var formViewModel = require('js/formViewModel');
+
+
+var SignInViewModel = oop.extend(formViewModel.FormViewModel, {
+    constructor: function () {
+        var self = this;
+        self.super.constructor();
+        self.username = ko.observable('').extend({
+            required: true,
+            email: true
+        });
+        self.password = ko.observable('').extend({
+            required: true,
+            minLength: 6,
+            maxLength: 35
+        });
+    },
+    isValid: function() {
+        var ValidationError = new formViewModel.ValidationError();
+        if (!this.username.isValid()) {
+            ValidationError.messages.push('Please enter a valid email address.');
+        }
+        if (!this.password.isValid()) {
+            ValidationError.messages.push('Your password must be more than six but fewer than 36 characters.');
+        }
+        if (ValidationError.messages.length > 0) {
+            throw ValidationError;
         } else {
-            return true;  // Allow form to submit normally
+            return true;
```

Comment:
According to your implementation of `FormViewModel`, returning `true` is unnecessary. I think this is fine, but returning `true` here might be misleading because it doesn't affect any behavior.